src/options.rs

```diff
@@ -132,20 +141,37 @@ impl Options {
       bail!("Bitcoin RPC server is on {rpc_chain} but ord is on {ord_chain}");
     }
 
+    let bitcoin_version = client.version()?;
+    if bitcoin_version < min_version {
+      bail!(
+        "Bitcoin Core {} or newer required, current version is {}",
+        Self::format_bitcoin_core_version(min_version),
+        Self::format_bitcoin_core_version(bitcoin_version),
+      );
+    }
+
     Ok(client)
   }
 
-  pub(crate) fn bitcoin_rpc_client_mainnet_forbidden(&self, command: &str) -> Result<Client> {
-    let client = self.bitcoin_rpc_client()?;
+  pub(crate) fn bitcoin_rpc_client_mainnet_forbidden(
+    &self,
+    command: &str,
+    min_version: usize,
+  ) -> Result<Client> {
+    let client = self.bitcoin_rpc_client(min_version)?;
 
     if self.chain() == Chain::Mainnet {
       bail!("`{command}` is unstable and not yet supported on mainnet.");
     }
     Ok(client)
   }
 
-  pub(crate) fn bitcoin_rpc_client_for_wallet_command(&self, command: &str) -> Result<Client> {
-    let client = self.bitcoin_rpc_client()?;
+  pub(crate) fn bitcoin_rpc_client_for_wallet_command(
```

Comment:
I think you're right about not passing the min version everywhere, let's only check the version in wallet commands, and they can all require the same min version.